### PR TITLE
Matrix improvements

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

A small step towards #23609 for matrices.

#### Other comments

I should add some tests.

#### Release Notes

* matrices
   * Better support for computing `adjoint`, `re`, and `im` for `Identity`, `OneMatrix`, and `ZeroMatrix`.

### CODE REVIEW

File: sympy/matrices/expressions/matexpr.py [hunk 1]
```diff
@@ -179,6 +179,9 @@ def _eval_conjugate(self):
         return Adjoint(Transpose(self))
 
     def as_real_imag(self, deep=True, **hints):
+        return self._eval_as_real_imag()
```
Comment (oscarbenjamin) [inline sympy/matrices/expressions/matexpr.py:182]: Shouldn't this method be deleted?

**Comment by asmeurer** [comment]
Interesting that the matrices already use _eval_as_real_imag, although it looks like the current implementation in master is wrong (see the `XXX` comment in common.py).

We might want to figure out for https://github.com/sympy/sympy/issues/23609 in general how we want to handle deprecation. It's hard to just switch from `method` to `_eval_method` because user subclasses that redefine `method` will break, because it will use that version instead of the base class version that calls `_eval_method`. 

So we need to either deprecate redefining `method`, or else somehow make it so it still works. But I'm not sure if that's possible.  I'm not even sure if we can issue a warning for redefining `method`, unless we put some logic in the metaclass.

**Comment by oscargus** [comment]
> Interesting that the matrices already use _eval_as_real_imag, although it looks like the current implementation in master is wrong (see the `XXX` comment in common.py).

Indeed! I was a bit confused about the status at first... Yes, I noted the comment but though that I take the simple way out for now and do not touch anything else.

> We might want to figure out for #23609 in general how we want to handle deprecation. It's hard to just switch from `method` to `_eval_method` because user subclasses that redefine `method` will break, because it will use that version instead of the base class version that calls `_eval_method`.

I guess that ideally, it shouldn't break anything but I can see many cases where it indeed may do it.

This is really a case where it would be nice to have code coverage... Right now, I modify things and hope that they are tested so that I do not break anything...

**Comment by oscargus** [comment]
I'd be a bit tempted to get this in 1.11 as well... But feel free to remove.

**Comment by oscargus** [comment]
The benefit is that in the added test, in current master e.g. re(X) becomes
image

while here it becomes
image

So basically more things involving re, im and adjoint evaluates to something more sensible.

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**